src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -99,10 +99,20 @@ private IntPtr AllocException(Exception ex)
         [DllImport("jitinterface")]
         private extern static char* GetExceptionMessage(IntPtr obj);
 
+#if READY_TO_RUN
```

Comment:
How about we set the sharing to the partial file plan? We'll probably still end up with a couple `#if READYTORUN` here and there, but having a field that is of different types between these two modes is a prime candidate for partial files.

We'll eventually have a bunch of methods that are only applicable to one of the three places `CorInfoImpl.cs` is included from.

ILCompiler.ReadyToRun\src\JitInterface\CorInfoImpl.ReadyToRun.cs can have:

```csharp
private ReadyToRunCodegenCompilation _compilation;

public CorInfoImpl(ReadyToRunCodegenCompilation compilation, JitConfigProvider jitConfig)
    // Call the common constructor that does the common initialization
    : this(jitConfig)
{
    _compilation = compilation;
}
```

Reply:
I broke out into CorInfoImpl.ReadyToRun.cs and CorInfoImpl.RyuJit.cs (included by ILCompiler.RyuJit and System.Private.Jit)